### ISSUE TITLE
Event source is not compatible with data produced by latest EVB where number of gains can depend on event type

### ISSUE THREAD

**Issue by sizun** [cta-observatory/ctapipe_io_nectarcam#15]
In the current _NectarCAMEventSource_, the number of gains _n_gains_ is an argument constructor and is assumed to be the same for all events. Also it could be derived from the _source.camera_config.nectarcam.algorithms_ parameter.

Since switching to version 4 of the event builder, this parameter is no longer filled as gain selection is enabled or not depending on the event type.

Maybe we should look how _LSTEventSource_ handled the transition.

**Comment by maxnoe** [comment]
See https://github.com/cta-observatory/ctapipe_io_lst/blob/e78acdc43ef4c4c95c83ec5100b7c6373af9df86/ctapipe_io_lst/__init__.py#L675

**Comment by jlenain** [comment]
Just to check, I think this issue has been addressed and is obsolete now, is that correct ?

**Comment by maxnoe** [comment]
Looking at the code in `__init__.py`, it seems like you correctly check on an event-by-event basis if the event was gain selected or not

**Comment by vmarandon** [comment]
Yes, but I would love to try on realistic data before closing this issue :-) 
(Including mixture of high-gain pixels and low-gain selected pixels)
